chronos/timer.nim

```diff
@@ -8,45 +8,141 @@
 #    Apache License, version 2.0, (LICENSE-APACHEv2)
 #                MIT license (LICENSE-MIT)
 
-
 ## This module implements cross-platform system timer with
 ## milliseconds resolution.
+##
+## Timer supports two types of clocks:
+## ``fast`` uses the most fast OS primitive to obtain wall clock time.
+## ``mono`` uses monotonic wall clack time (default)
+##
+## You can specify which timer you want to use ``-d:asyncTimer=<fast/mono>``.
+const asyncTimer* {.strdefine.} = "mono"
 
 when defined(windows):
+  when asyncTimer == "fast":
+    from winlean import DWORD, getSystemTimeAsFileTime, FILETIME
+
+    proc fastEpochTime*(): uint64 {.inline.} =
+      ## Timer resolution is millisecond.
+      var t: FILETIME
+      getSystemTimeAsFileTime(t)
+      result = ((cast[uint64](t.dwHighDateTime) shl 32) or
+                 cast[uint64](t.dwLowDateTime)) div 10_000
+
+    proc fastEpochTimeNano*(): uint64 {.inline.} =
+      ## Timer resolution is nanosecond.
+      var t: FILETIME
+      getSystemTimeAsFileTime(t)
+      result = ((cast[uint64](t.dwHighDateTime) shl 32) or
+                 cast[uint64](t.dwLowDateTime)) * 100
+
+  else:
+    proc QueryPerformanceCounter(res: var uint64) {.
+      importc: "QueryPerformanceCounter", stdcall, dynlib: "kernel32".}
+    proc QueryPerformanceFrequency(res: var uint64) {.
+      importc: "QueryPerformanceFrequency", stdcall, dynlib: "kernel32".}
+
+    var queryFrequencyM: uint64
+    var queryFrequencyN: uint64
+
+    proc fastEpochTimeNano*(): uint64 {.inline.} =
```

Comment:
so let's introduce a type `Moment` for the monotonic "current time". we can also introduce an api like:
```
proc now(T: type Moment): Moment = ...
```
which makes it explicit what kind of time we're dealing with, at the call site:
```
let moment = Moment.now()
```

Reply:
This will be next step.

Reply:
sure. when we have that, do we still want to expose the public name `fastEpochTimeNano`?